pkg/platform/kube/functionres/lazy.go

```diff
@@ -299,6 +302,21 @@ func (lc *lazyClient) WaitAvailable(ctx context.Context, namespace string, name
 				break
 			}
 		}
+
+		// get the deployment pods. if it doesn't exist yet, retry a bit later
+		pods, err := lc.kubeClientSet.CoreV1().
+			Pods(namespace).
+			List(metav1.ListOptions{
+				LabelSelector: common.CompileListFunctionPodsLabelSelector(name),
+			})
+		if err != nil {
+			continue
+		}
+
+		if functionDeploymentFailed, err := lc.isFunctionDeploymentFailed(pods.Items,
```

Comment:
worth checking what is happening here when deploying a function in "disabled" mode (meaning, no pods would be spawn)

Reply:
If no pods will be spawned, then listing pod items (line _307_ above) will retrieve an empty list, and `isFunctionDeploymentFailed` will return `false`.